sklearn/naive_bayes.py

```diff
@@ -344,9 +377,24 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
         # deviation of the largest dimension.
         epsilon = 1e-9 * np.var(X, axis=0).max()
 
+        # reset flag if refit called
         if _refit:
             self.classes_ = None
 
+        # set classes- initialized values/passed in partial_fit/infer from y
```

Comment:
It looks like GNB may be a bit of an outlier in not using LabelEncoder for this.

We need a way to rely on helpers, not ad hoc code in each estimator.

I also suspect we'll want to deprecate the classes parameter to partial_fit